src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.js

```diff
@@ -0,0 +1,272 @@
+import Str from 'expensify-common/lib/str';
+import lodashGet from 'lodash/get';
+import React, {Component} from 'react';
+import {View, ScrollView, TouchableOpacity} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import Navigation from '../../../../libs/Navigation/Navigation';
+import ROUTES from '../../../../ROUTES';
+import ScreenWrapper from '../../../../components/ScreenWrapper';
+import HeaderWithCloseButton from '../../../../components/HeaderWithCloseButton';
+import compose from '../../../../libs/compose';
+import ONYXKEYS from '../../../../ONYXKEYS';
+import withLocalize, {withLocalizePropTypes} from '../../../../components/withLocalize';
+import MenuItem from '../../../../components/MenuItem';
+import styles from '../../../../styles/styles';
+import * as Expensicons from '../../../../components/Icon/Expensicons';
+import Text from '../../../../components/Text';
+import OfflineWithFeedback from '../../../../components/OfflineWithFeedback';
+import ConfirmModal from '../../../../components/ConfirmModal';
+import * as User from '../../../../libs/actions/User';
+import TextInput from '../../../../components/TextInput';
+import CONST from '../../../../CONST';
+import Icon from '../../../../components/Icon';
+import colors from '../../../../styles/colors';
+import Button from '../../../../components/Button';
+import * as ErrorUtils from '../../../../libs/ErrorUtils';
+import themeColors from '../../../../styles/themes/default';
+import FullScreenLoadingIndicator from '../../../../components/FullscreenLoadingIndicator';
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** Login list for the user that is signed in */
+    loginList: PropTypes.shape({
+        /** Value of partner name */
+        partnerName: PropTypes.string,
+
+        /** Phone/Email associated with user */
+        partnerUserID: PropTypes.string,
+
+        /** Date when login was validated */
+        validatedDate: PropTypes.string,
+
+        /** Field-specific server side errors keyed by microtime */
+        errorFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+
+        /** Field-specific pending states for offline UI status */
+        pendingFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+    }),
+
+    /** Current user session */
+    session: PropTypes.shape({
+        email: PropTypes.string.isRequired,
+    }).isRequired,
+
+    /** Route params */
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            /** passed via route /settings/profile/contact-methods/:contactMethod/details */
+            contactMethod: PropTypes.string,
+        }),
+    }),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    loginList: {},
+    route: {
+        params: {
+            contactMethod: '',
+        },
```

Comment:
The problem I see with that is then you get this error:

<img width="839" alt="image" src="https://user-images.githubusercontent.com/87341702/225739824-51f41282-ce53-4f4b-9624-33f3943ec9e8.png">

And I'm not sure that making `route/params/contactMethod` non required is the best, considering that it is required?

Reply:
Aah yeah that's interesting, I am not sure it's beneficial in this situation to use `initialParams` - but this could be an interesting discussion if there's some drawback of the current solution? Otherwise I think we should keep what we have as @aldo-expensify said